src/clients/SpokePoolClient.ts

```diff
@@ -334,6 +336,24 @@ export class SpokePoolClient extends BaseAbstractClient {
     return fills?.find((fill) => validateFillForDeposit(fill, deposit));
   }
 
+  private isEvmRepaymentValid(fill: Fill, matchedDeposit: Deposit): boolean {
+    // Slow fills don't result in repayments so they're always valid.
+    if (isSlowFill(fill)) {
+      return true;
+    }
+    // Lite chain deposits force repayment on origin chain.
+    const repaymentChainId = matchedDeposit.fromLiteChain ? fill.originChainId : fill.repaymentChainId;
+    // Return undefined if the requested repayment chain ID is not recognized by the hub pool.
+    const possibleRepaymentChainIds = this.configStoreClient?.getEnabledChains();
+    if (isDefined(possibleRepaymentChainIds) && !possibleRepaymentChainIds.includes(repaymentChainId)) {
+      return false;
+    }
+    if (!chainIsEvm(repaymentChainId)) {
+      return false;
+    }
+    return isValidEvmAddress(fill.relayer);
```

Comment:
Small nit - can maybe squash this a bit. I think this helps readability.

```suggestion
    return chainIsEvm(repaymentChainId)) && isValidEvmAddress(fill.relayer);
```